### ISSUE TITLE
Strange Bug - A book in the database is not found

### ISSUE THREAD

**Issue by jaschaurbach** [bookwyrm-social/bookwyrm#2587]
I searched for a book and imported it. After the import I got a "page not found error". 

When I search now for the book, I get it as present on my instance 

<img width="1163" alt="Bildschirm­foto 2023-01-10 um 18 10 31" src="https://user-images.githubusercontent.com/2017105/211617562-209dd0eb-0966-4f81-b219-0a0fc9046ad0.png">

When I click the link I get a not found error:

<img width="1163" alt="Bildschirm­foto 2023-01-10 um 18 10 37" src="https://user-images.githubusercontent.com/2017105/211617656-e835142c-28b0-4f59-9540-ae615d399628.png">

The log just shows the following:
```
2023-01-10 18:10:33,734 DEBG 'bookwyrm' stderr output:
Not Found: /book/12575/s/the-speed-book
```

This happened after I pulled the production branch with the new commits. Any ideas?

**Comment by marcusyoung** [comment]
I've had this happen to a user on v0.5.3. Was an import from bookwyrm.social. In addition if you try to import it again it also goes to the 404 error. To fix I had to delete the book from three database tables and then import again. Then it imported fine.

**Comment by mouse-reeve** [comment]
I suspect what's happening is that the book didn't import properly, and is missing its associated work; that will cause a 404. So there are a couple related bugs here: one is that the book was created improperly, and the other is that the application should be able to repair books that are missing works

**Comment by jaschaurbach** [comment]
We really need “Book-Edit” feature on the front end. I know, it I not trivial a needs time and someone to do it.

but we really need it.

**Comment by mouse-reeve** [comment]
I believe this was fixed by #2905
